pkgs/tools/networking/wireguard-go/default.nix

```diff
@@ -0,0 +1,27 @@
+{ stdenv, buildGoPackage, fetchurl }:
+
+buildGoPackage rec {
+  name = "wireguard-go-${version}";
+  version = "0.0.20180514";
+
+  goPackagePath = "wireguard-go";
+
+  src = fetchurl {
+    url = "https://git.zx2c4.com/wireguard-go/snapshot/wireguard-go-${version}.tar.xz";
+    sha256 = "1bn49a67m2ab0l9lq3zh2mfbbppmyg34klqi3069sjn6lg2hlajs";
+  };
+
+  goDeps = ./deps.nix;
+
+  patchPhase = ''
+    # Replace local imports so that go tools do not trip on them
+    find . -name '*.go' -exec sed -i '/import (/,/)/s@"./@"${goPackagePath}/@' {} \;
+  '';
+
+  meta = with stdenv.lib; {
+    description = "Userspace Go implementation of WireGuard";
+    homepage = https://git.zx2c4.com/wireguard-go/about/;
+    license = licenses.gpl2;
+    platforms = with platforms; linux ++ darwin ++ windows;
```

Comment:
`buildGoPackage` sets `go.meta.platforms` to all Go packages, and that includes Linux and Darwin: https://github.com/NixOS/nixpkgs/blob/6a2932717585f58867fce4376434eaf54cf604b8/pkgs/development/go-modules/generic/default.nix#L218

Reply:
I’m not sure what to make of this. It sets some default platforms for those who are lazy, but we want Windows as well, don’t we? I mean, wireguard-go explicitly supports windows and the fact that `go` in `nixpkgs` does not is, well, a separate matter...

Reply:
OK, you are right.